tokenserver/assignment/sqlnode/sql.py

```diff
@@ -586,14 +586,20 @@ def add_node(self, service, node, capacity, **kwds):
         # We release only a fraction of the node's capacity to start.
         if available is None:
             available = math.ceil(capacity * self.capacity_release_rate)
-        res = self._safe_execute(sqltext(
-            """
-            insert into nodes (id, service, node, available, capacity,
-                            current_load, downed, backoff)
-            values (:nodeid, :service, :node, :available, :capacity,
-                    :current_load, :downed, :backoff)
-            """),
-            nodeid=kwds.get('nodeid', None),
+        cols = ["service", "node", "available", "capacity",
+                "current_load", "downed", "backoff"]
+        args = [":" + v for v in cols]
+        # Handle test cases that require nodeid to be 800
+        if "nodeid" in kwds:
+            cols.append("id")
+            args.append(":nodeid")
```

Comment:
I'm not sure why this didn't work with just allowing `nodeid` to default to `None` and relying on autoincrement to do its thing, but I have no objection to making it more explicit here. There's probably some sqlalchemy query-constructor thing that would save you a bit of string interpolation here, but it would take you too far away from the main point of this PR. So 👍 .

Reply:
I've kinda learned that it's best not to specify values when no value is desired. Sometimes databases or ORMs can be "helpful" and force `None` instead of just ignoring the value. I'm fine doing the extra work here so that we're very clear about what's going on. 

I think if we were to use the sqlalchemy constructor route, we'd essentially be doing the same thing.